pkg/common/auth_credentials/auth_credentials.go

```diff
@@ -0,0 +1,74 @@
+package auth_credentials
+
+import (
+	"fmt"
+	envoyServiceAuthV3 "github.com/envoyproxy/go-control-plane/envoy/service/auth/v3"
+	"regexp"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"strings"
+)
+
+type AuthCredentials interface {
+	GetCredentialsFromReq(*envoyServiceAuthV3.AttributeContext_HttpRequest) (string, error)
```

Comment:
This `envoyServiceAuthV3.AttributeContext_HttpRequest` is spreading within our code like covid. On another PR, I was thinking about replacing with sth like: https://github.com/projectcontour/contour-authserver/blob/main/pkg/auth/convert.go